src/extension/background-script/actions/alby/enable.ts

```diff
@@ -38,8 +40,14 @@ const enable = async (message: MessageAllowanceEnable, sender: Sender) => {
           if (!allowance.id) {
             return { data: { error: "id is missing" } };
           }
+
+          if (!enabledFor.has("alby")) {
```

Comment:
@pavanjoshi914 this check is not needed any more since we are using a set now

Reply:
why is it different if we use set there? it's done to avoid duplicate entries in enabledFor. 

for example, enable the allowance. we will have entry. and then lock the extension and then enable the allowance again using same provider. see duplicate entries. next time we update the allowance we add duplicate entries. i think this is still needed. kindly cross check again

Reply:
@pavanjoshi914 it uses the javascript `Set` class. Which is a collection without duplicates.

Reply:
ohh okay! i haven't used set before thanks for the change